### PR TITLE
event listeners

### PR DESCRIPTION
**Pull request by Baggerone**
Is this a good way to structure things?
(Sorry, I couldn't resist tweaking a few things that weren't really related)

### CODE REVIEW

File: application/models/user_access_token.go [hunk 1]
```diff
@@ -159,15 +158,37 @@ func (u *UserAccessToken) GetUser() (User, error) {
...
+			return true, fmt.Errorf("unable to delete expired userAccessToken, id: %v", u.ID)
 		}
-		return true
+		return true, nil
 	}
-	return false
+	return false, nil
+}
+
+func UserAccessTokensDeleteExpired() (int, error) {
```
Comment (Schparky) [inline application/models/user_access_token.go]:
```suggestion
func (u *UserAccessTokens) DeleteExpired() (int, error) {
```

File: application/models/user_access_token_test.go [hunk 2]
```diff
@@ -520,7 +524,95 @@ func (ms *ModelSuite) TestUserAccessToken_Renew() {
...
+	}
+
+	gotToken := uats[0].AccessToken
+	wantToken := f.UserAccessTokens[1].AccessToken
+
+	if gotToken != wantToken {
+		t.Errorf("Wrong token remaining. \nWant %v \n Got %v", wantToken, gotToken)
+	}
+
+}
```
Comment (Schparky) [inline application/models/user_access_token_test.go]: suggest using `ms.Equal` for these comparisons

File: application/models/user_access_token_test.go [hunk 3]
```diff
@@ -520,7 +524,95 @@ func (ms *ModelSuite) TestUserAccessToken_Renew() {
...
+	ResetTables(t, ms.DB)
+
+	f := createFixtures_UserAccessTokensDeleteExpired(ms, t)
+
+	got, err := UserAccessTokensDeleteExpired()
+
+	if err != nil {
+		t.Errorf("Unexpected error: %v", err)
+		return
+	}
```
Comment (Schparky) [inline application/models/user_access_token_test.go]: `ms.NoError(err)` is a concise way to do this

File: application/actions/auth.go [hunk 4]
```diff
@@ -406,10 +406,20 @@ func SetCurrentUser(next buffalo.Handler) buffalo.Handler {
 
 		var userAccessToken models.UserAccessToken
 		err := userAccessToken.FindByBearerToken(bearerToken)
-		if err != nil || userAccessToken.IsExpired() {
+		if err != nil {
+			domain.ErrLogger.Print(err.Error())
 			return c.Error(401, fmt.Errorf("invalid bearer token"))
 		}
 
+		isExpired, err := userAccessToken.DeleteIfExpired()
```
Comment (Schparky) [inline application/actions/auth.go]: I like the function name change.

File: application/domain/domain.go [hunk 5]
```diff
@@ -24,14 +26,30 @@ const (
 	EmptyUUID                  = "00000000-0000-0000-0000-000000000000"
 	DateFormat                 = "2006-01-02"
 	MaxFileSize                = 1 << 20 // 1 Mebibyte
-	UIURLEnv                   = "UI_URL"
 	AccessTokenLifetimeSeconds = 3600
+	DateTimeFormat             = "2006-01-02 15:04:05"
+
+	// Environment Variables
```
Comment (Schparky) [inline application/domain/domain.go]: Good groupings. I've seen this done with new `const` blocks. Thoughts on that?

File: application/listeners/listeners.go [hunk 6]
```diff
@@ -0,0 +1,71 @@
+package listeners
+
+import (
+	"github.com/gobuffalo/events"
+	"github.com/silinternational/wecarry-api/domain"
+	"github.com/silinternational/wecarry-api/models"
+	"time"
+)
+
+// For new listener functions, register them at the end of the
```
Comment (Schparky) [inline application/listeners/listeners.go]: Why not just put the slice here?

Comment (Baggerone) [reply application/listeners/listeners.go]: Ha, ha.  That's my Python experience showing through.

File: application/models/user_access_token_test.go (hunk 3 above)
Comment (Baggerone) [reply application/models/user_access_token_test.go]: Done.

File: application/models/user_access_token_test.go (hunk 2 above)
Comment (Baggerone) [reply application/models/user_access_token_test.go]: Good, thanks.

File: application/models/user_access_token.go (hunk 1 above)
Comment (Baggerone) [reply application/models/user_access_token.go]: Done.

**Comment by Baggerone** [comment]
I think it's preferable leaving the Loggers as globals rather than in a context that gets passed to all the functions.  I see them as more like constants.  Once we set their output in the init(), we're only using their methods rather than modifying their attributes.

File: application/domain/domain.go (hunk 5 above)
Comment (Baggerone) [reply application/domain/domain.go]: Sounds good.

**Pull request merged**